### PR TITLE
fix: hide metadata section if there's no metadata

### PR DESCRIPTION
**Pull request by keita-determined**
## Description

- Hide `Metadata` section if there's no data to show

## Test Plan

- Check if metadata section appears if metadata exits
- Check if metadata section is hidden if no metadata exits

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/Metadata.tsx [hunk 1]
```diff
@@ -51,6 +50,11 @@ const Metadata: React.FC<Props> = ({ trial }: Props) => {
 
   const treeData = (trial?.metadata && getNodes(trial?.metadata)) ?? [];
 
+  if (treeData.length === 0) {
```
Comment (ashtonG) [inline webui/react/src/components/Metadata.tsx:53]: can we move the check for metadata up to the trial details overview?

Comment (keita-determined) [reply webui/react/src/components/Metadata.tsx]: fixed

**Review by ashtonG — APPROVED** [review]

File: webui/react/src/components/Metadata.tsx
Comment (EmilyBonar) [inline webui/react/src/components/Metadata.tsx:1]: nit: you can change the `trial` param to be required now that this component is gated behind the metadata check

Comment (keita-determined) [reply webui/react/src/components/Metadata.tsx:1]: fixed

**Pull request merged**